releng/docker/Dockerfile

```diff
@@ -7,34 +9,99 @@
 # repository you want on /zbm inside the container.
 
 # Use the official Void Linux container
-FROM ghcr.io/void-linux/void-glibc-full:latest
+FROM ghcr.io/void-linux/void-glibc-full 
+LABEL org.opencontainers.image.authors="ZFSBootMenu Team, https://zfsbootmenu.org"
+# ZFSBootMenu commit hash, tag or branch name used by
+# default to build ZFSBootMenu images (default: master)
 ARG ZBM_COMMIT_HASH
+# Include the specified Void Linux package in the image
+#
+# (multiple entries must be seperated by spaces)
+ARG PACKAGES=
+ARG XBPS_REPOS="https://repo-fastly.voidlinux.org/current"
+# Include the specified Void Linux kernel series in the image; kernel
+# series take the form <major>.<minor> and correspond to Void packages
+# linux<kver> and linux<kver>-headers
+#
+# Default: install 5.10, 5.15 and 6.1
+#	
+# (multiple entries must be seperated by spaces)
+ARG KERNELS="linux5.10 linux5.15 linux6.1" 
+
+RUN mkdir -p /etc/xbps.d
+RUN <<EOF
+for repo in "${XBPS_REPOS}"; do
+  echo "repository=$repo" >> /etc/xbps.d/00-custom-repos.conf
+done
+EOF
 
 # Ensure everything is up-to-date
-RUN xbps-install -Suy xbps && xbps-install -uy
+RUN xbps-install -Suy xbps && xbps-install -uy bash
+SHELL ["/bin/bash", "-c"]
+
+# Install custom packages
+RUN <<EOF
+if [ -n "${PACKAGES}" ]; then
+  	echo "INFO: adding extra Void packages: ${PACKAGES}"
+	xbps-install -y $PACKAGES
+fi
+EOF
 
 # Prefer an LTS version over whatever Void thinks is current
-RUN echo "ignorepkg=linux" > /etc/xbps.d/10-nolinux.conf \
-	&& echo "ignorepkg=linux-headers" >> /etc/xbps.d/10-nolinux.conf
+RUN <<EOF
+echo "ignorepkg=linux" >> /etc/xbps.d/10-nolinux.conf
+echo "ignorepkg=linux-headers" >> /etc/xbps.d/10-nolinux.conf
+EOF
 
-# Install components necessary to build the image
+# Prevent initramfs kernel hooks from being installed
+RUN <<EOF
+echo "noextract=/usr/libexec/mkinitcpio/*" >> /etc/xbps.d/15-noinitramfs.conf
+echo "noextract=/usr/libexec/dracut/*" >> /etc/xbps.d/15-noinitramfs.conf
+EOF
+
+# Install ZFSBootMenu dependencies and components necessary to build images
 RUN xbps-query -Rp run_depends zfsbootmenu | xargs xbps-install -y
-RUN xbps-install -y linux5.10 linux5.10-headers \
-	zfs gummiboot-efistub curl yq-go bash kbd \
-	terminus-font dracut mkinitcpio cryptsetup
+RUN <<EOF
+if [ -n "${KERNELS}" ]; then
+	for _kern in "${KERNELS}"; do
+		kern_headers+=( "${_kern}-headers" )
+	done
+	echo Installing Kernels: ${KERNELS} ${kern_headers}
+	xbps-install -y ${KERNELS} ${kern_headers}
+fi
```

Comment:
If `$KERNELS` is empty, it defeats the whole purpose of building the container. This probably doesn't need to be wrapped in a test, although it would be nice to fail the build up front if $KERNELS is unspecified.

Also, `for _kern in "${KERNELS}"` is probably wrong; in regular shell, it would treat `linux6.1 linux5.10 linux5.15` as a single argument instead of three. Unless the build process is running this in some non-standard shell-like environment, it must not be quoted.